src/rgw/rgw_sync_module_aws.cc

```diff
@@ -972,6 +971,17 @@ int RGWAWSSyncModule::create_instance(CephContext *cct, map<string, string, ltst
   if (i != config.end())
     conf.s3_endpoint = i->second;
 
+  i = config.find("host_style");
+  string host_style_str;
+  if (i != config.end())
+    host_style_str = i->second;
+
+  if (host_style_str == "" || host_style_str != "virtual") {
```

Comment:
no need for two checks here, can just `if (host_style_str != "virtual")`, unless you change the second check to `if (host_style_str == "path")`